### ISSUE TITLE
Persistence: previousState(true) returns wrong HistoricItem

### ISSUE THREAD

**Issue by Jogobo** [openhab/openhab-core#3472]
I am on openHab 3.4.2 and I am using jdbc:mariadb for persistence. I have a temperature item from which I want to fetch the previous state different from current state with "Item.previousState(true).state". I was surprised that I received the same value as the current state so I started to investigate.
The reason for the "wrong" previous state seems to be the dimension of the item. So I tested the following:

**Item Type=Number**
Database entries in item table:
```
'2023-03-21 16:20:09' 20
'2023-03-21 07:31:30' 20
'2023-03-21 06:02:14' 19
```

```
Item.state = 20
Item.previousState() = state=20.0, timestamp=2023-03-21T16:20:09+01:00[Europe/Berlin]
Item.previousState(true) = state=19.0, timestamp=2023-03-21T06:02:14+01:00[Europe/Berlin]
```

**Item Type=Number:Temperature**
Database entries in item table:
```
'2023-03-21 16:25:08' 20
'2023-03-21 16:20:09' 20
'2023-03-21 07:31:30' 20
'2023-03-21 06:02:14' 19
```

```
Item.state = 20 °C
Item.previousState() = state=20.0, timestamp=2023-03-21T16:25:08+01:00[Europe/Berlin]
Item.previousState(true) = state=20.0, timestamp=2023-03-21T16:25:08+01:00[Europe/Berlin]
```

It looks like the comparison of equality between current state and previous state is not based on the base type "Number" but on the current state with unit.

My current workaround is 
`Item.historicState(Item.lastUpdate.minusSeconds(1)).state`

**Comment by J-N-K** [comment]
```
logInfo("x1", T1.state.toString)
logInfo("x1", T1.previousState.toString)
logInfo("x1", T1.previousState(true).toString)
```

```
19:15:10.382 | INFO  | qtp1886515295-365    | org.openhab.core.model.script.x1     | 199 | 19 °C
19:15:10.387 | INFO  | qtp1886515295-365    | org.openhab.core.model.script.x1     | 199 | JdbcItem [name=T1, state=19 °C, timestamp=2023-03-22T19:05:32.869+01:00[Europe/Berlin]]
19:15:10.391 | INFO  | qtp1886515295-365    | org.openhab.core.model.script.x1     | 199 | JdbcItem [name=T1, state=20 °C, timestamp=2023-03-22T19:03:35.420+01:00[Europe/Berlin]]
```

Looks good to me.

**Comment by Jogobo** [comment]
```
+---------------------+-------+
| Time                | Value |
+---------------------+-------+
| 2023-03-22 19:02:42 |    19 |
| 2023-03-22 18:25:34 |    18 |
| 2023-03-22 17:31:09 |    19 |
| 2023-03-22 07:30:01 |    20 |
+---------------------+-------+
```
```
Item.state: 19 °C
Item.previousState(): JdbcItem [name=Item, state=19.0, timestamp=2023-03-22T19:02:42+01:00[Europe/Berlin]]
Item.previousState(true): JdbcItem[name=Item, state=19.0, timestamp=2023-03-22T19:02:42+01:00[Europe/Berlin]]
```
I don't know what "toString" does in this context, but the Item.state is different from HistoricItem.state.

```
logInfo("Persistence", "state == previousState().state: " + (Item.state == Item.previousState().state)
state == previousState().state: false
```
```
logInfo("Persistence", "state == previousState(true).state: " + (Item.state == Item.previousState(true).state)
state == previousState(true).state: false
```
comparison results in "false" for previousState() and previousState(true).

**Comment by J-N-K** [comment]
Can you please show the result of `Item.getUnit.toString`? It seems that no unit could be retrieved from the item. The code in JDBC persistence is unchanged since 01/2021.

**Comment by Jogobo** [comment]
getUnit returns null. Though Item.state shows "19 °C"

**Comment by J-N-K** [comment]
Then the issue is your configuration. Do you have a state description configured?

**Comment by Jogobo** [comment]
Don't know what you mean by "state description". Here is the item config:
![image](https://user-images.githubusercontent.com/62647003/227033304-d0358888-9f51-40e6-9eb0-ae4c2379607b.png)

**Comment by J-N-K** [comment]
On the page "above that", is there any metadata configured?

<img width="1193" alt="Bildschirm­foto 2023-03-22 um 21 45 54" src="https://user-images.githubusercontent.com/5159123/227034117-7b4c90df-7210-45aa-8cc5-929d442798cb.png">

**Comment by Jogobo** [comment]
Yes, for Alexa.
![image](https://user-images.githubusercontent.com/62647003/227034378-d4162942-0500-4457-afe4-ea7e61df8cef.png)

**Comment by J-N-K** [comment]
On that page, is there a value with the unit displayed or not?

**Comment by J-N-K** [comment]
Try adding additional metadata "state description", add `%.2f °C` as pattern.

**Comment by Jogobo** [comment]
Adding "state description" to the item changed the states of the HistoricItems returned by previousState() and previousState(true). The unit is added to the historic state.

```
2023-03-22 22:08:00.788 [INFO ] [penhab.core.model.script.Persistence] - current: 18 °C, 
prev(): JdbcItem [name=ArbeitszimmerINT0000004_1_SetTemperature, state=18 °C, timestamp=2023-03-22T22:00:45+01:00[Europe/Berlin]], 
prev(true): JdbcItem [name=ArbeitszimmerINT0000004_1_SetTemperature, state=19 °C, timestamp=2023-03-22T21:56:27+01:00[Europe/Berlin]]
2023-03-22 22:08:00.794 [INFO ] [penhab.core.model.script.Persistence] - state == previousState().state: true
2023-03-22 22:08:00.859 [INFO ] [penhab.core.model.script.Persistence] - state == previousState(true).state: false
2023-03-22 22:08:00.862 [INFO ] [penhab.core.model.script.Persistence] - getUnit: °C
```

Do I have to add "state description" to all items in my persistence strategy?

**Comment by J-N-K** [comment]
No, the reason is that a state description is provided by the binding that does not allow to determine the unit (probably `%unit%`). This can't be fixed in OH 3.4.x due to a design flaw that links display state and internal representation. This will be fixed with the solution to #3282 in OH4.
